subsys/testsuite/ztest/src/ztest_defaults.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ztest_test_new.h"
+
+/**
+ * @brief Try to shorten a filename by removing the current directory
+ *
+ * This helps to reduce the very long filenames in assertion failures. It
+ * removes the current directory from the filename and returns the rest.
+ * This makes assertions a lot more readable, and sometimes they fit on one
+ * line.
+ *
+ * @param file Filename to check
+ * @returns Shortened filename, or @file if it could not be shortened
+ */
+const char *ztest_relative_filename(const char *file)
+{
+	return file;
+}
```

Comment:
This is already the default implementation in ztest.c, not needed.

Reply:
When `ZTEST_NEW_API` is enabled ztest.c is not compiled leaving a undefined reference when linking for legacy configurations.  I did tinker around a bit to not have this duplicated but it looked messy keeping legacy, ZTEST_NEW POSIX, and ZTEST_NEW !POSIX configs happy.  This looked a bit cleaner.